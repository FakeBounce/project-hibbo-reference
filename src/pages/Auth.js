import React from 'react';

import AuthComponent from '../components/auth';

const Auth = props => {
  return (
    <AuthComponent {...props} />
  );
};

export default Auth;
