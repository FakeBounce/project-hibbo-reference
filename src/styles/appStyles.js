/* eslint global-require: 0 */
import { Dimensions } from 'react-native';

import { iconVideos } from 'utils/storedData';

const { width } = Dimensions.get('window');

export default {
  prefix: {
    fixedIcon: 'icon_opac/',
  },
  event: {
    none: 0,
    gateau: 1,
    sapin: 2,
    oeuf: 3,
    souris: 4,
    bonbon: 11,
    manette: 12,
    moto: 13,
    voiture: 14,
    cadeau_vert: 15,
    cadeau_bleu: 16,
    courses: 21,
    rangement: 22,
    crayon: 23,
    marteau: 24,
    lapin: 31,
    chat: 32,
    chien: 33,
    heureux: 41,
    betise: 42,
    system: {
      reset: 0xffff,
      setLocal: 0xfff0,
      setVolume: 0xfff1,
      setObjectif: 0xfff2,
    },
  },
  icons: iconVideos,
  maxTextSize: 70,
  colors: {
    white: '#FFF',
    black: '#000',
    darkGrey: '#303030',
    darkBlue: '#06263D',
    darkBlueModal: '#06263D',
    grey: '#979797',
    greyOpac: 'rgba(151, 151, 151, 0.18)',
    whiteOpac: 'rgba(255, 255, 255, 0.30)',
    green: '#16EBC5',
    fb: '#4100C0',
    google: '#D5492F',
    primary: '#413D3D',
    textGrey: '#A7A7A7',
    buttonGrey: '#EFEFEF',
    lightGrey: '#9B9B9B',
    lightGreyBullet: '#E5E3DD',
    blueGreen: '#16EBC5',
    borderGrey: 'rgba(74, 74, 74, 0.23)',
    blackModal: 'rgba(0, 0, 0, 0.5)',
    toggleButtonGrey: '#4A4A4A',
    yellow: '#FFDF2D',
    moneyPotText: '#FFFE82',
    buttonYellow: '#FFFE82',
    cardBlue: '#697A99',
    gold: '#FEE9BF',
    red: '#FC5151',
    error: '#FF6969',
    textLink: '#687A99',
    messageGrey: '#DDE6EC',
    blue: '#0084ff',
    balanceGreen: '#3CDD80',
    pelicanColor: '#A685FF',
    deepBlue: '#028DAF',
    flatPink: '#FF6682',
    flatOrange: '#FFB434',
    flatPurple: '#82438F',
    white2pac: 'rgba(255,255,255,0.2)',
    white1pac: 'rgba(255,255,255,0.1)',
    placeholderGrey: 'rgba(229, 227, 221, 0.5)',
    greenStrong: '#50E3C2',
    orangeNotStrong: '#F5A623',
    inputLabel: '#C3C3C3',
    inputGrey: '#E8E8E8',
  },
  transaction: {
    minimunForAnimation: 1000,
  },
  time: {
    moneyTransaction: 2500,
  },
  textSizes: {
    xxxsmall: 10,
    xxsmall: 13,
    xsmall: 16,
    small: 18,
    regular: 22,
    normal: 24,
    xnormal: 26,
    big: 36,
    xbig: 50,
    xxbig: 58,
  },
  size: {
    cardMargin: 10,
    maxBubbleWidth: 190,
  },
  cardStyle: {
    shadowOpacity: 0,
    shadowColor: 'transparent',
    borderRadius: 10,
  },
  fonts: {
    circularBook: 'Circular-Std-Book',
    circularBookItalic: 'Circular-Std-Book-Italic',
    circularMedium: 'Circular-Std-Medium',
    circularMediumItalic: 'Circular-Std-Medium-Italic',
    circularBold: 'Circular-Std-Bold',
    circularBoldItalic: 'Circular-Std-Bold-Italic',
    circularBlackItalic: 'Circular-Std-Black-Italic',
    pixel: 'LCDDot-TR',
    chewy: 'Chewy',
    peepo: 'Peepo',
  },
  get colorList() {
    return [
      this.colors.deepBlue,
      this.colors.flatPink,
      this.colors.flatOrange,
      this.colors.flatPurple,
    ];
  },
  colorListSuffix: ['_bleu', '_rose', '_orange', '_violet'],
  swiperRange: [0, width / 2, width, width * 1.5, width * 2],
};
