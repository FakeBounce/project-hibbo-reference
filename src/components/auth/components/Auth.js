import React, { PureComponent } from 'react';
import { Animated, Dimensions, Easing } from 'react-native';
import { func, shape, string, bool } from 'prop-types';
import { bind } from 'decko';
import { NavigationActions } from 'react-navigation';
import { getLanguage, getTranslations } from 'utils/i18n';
import shallowequal from 'shallowequal';
import {
  StyledContainerBlack,
  StyledAndroidWrapper,
} from 'styledComponents/containers';

import appStyles from 'styles/appStyles';
import Visual from './Visual';
import SelectAuthProvider from './SelectAuthProvider';
import Register from './Register';
import Login from './Login';
import ResetPassword from './ResetPassword';
import FormValidationHOC from './FormValidationHOC';
import TermsOfServices from './Terms/TermsOfServices';

import * as Styles from '../styles';

let RegisterExtanded = null;
let LoginExtanded = null;
let ResetPwdExtanded = null;

class Auth extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      authType: '',
      authProvider: '',
      isButtonDisabled: false,
      step: 0,
      controlsShown: true,
      animRunning: false,
      imageContainer: {
        translateY: new Animated.Value(0),
      },
      image: {
        transform: [
          {
            translateY: new Animated.Value(0),
          },
        ],
      },
      controls: {
        height: new Animated.Value(0),
        opacity: new Animated.Value(0),
      },
      form: {
        opacity: new Animated.Value(0),
      },
      visual: {
        opacity: new Animated.Value(1),
      },
      modalStyle: {
        opacity: new Animated.Value(0),
      },
    };

    this.editUserProfile = this.props.editUserProfile.bind(this);
    this.startRegister = this.props.startRegister.bind(this);
    this.registerWithGoogle = this.props.registerWithGoogle.bind(this);
    this.loginWithGoogle = this.props.loginWithGoogle.bind(this);
    this.registerWithFb = this.props.registerWithFb.bind(this);
    this.loginWithFb = this.props.loginWithFb.bind(this);
    this.resetAllErrors = this.props.resetAllErrors.bind(this);
  }

  componentDidMount() {
    const register = {
      label: [
        [getTranslations('auth.label.email'), 'email'],
        [getTranslations('auth.label.lastname'), 'lastname'],
        [getTranslations('auth.label.firstname'), 'firstname'],
        [getTranslations('auth.label.phone'), 'phone'],
        [getTranslations('auth.label.password'), 'password'],
      ],
      form: {
        email: '',
        firstname: '',
        lastname: '',
        phone: '',
        password: '',
        language: getLanguage(),
        homeAddress: {
          line1: '',
          city: '',
          region: '',
          postalCode: '',
          country: '',
        },
      },
    };
    const login = {
      label: [
        [getTranslations('auth.label.email'), 'email'],
        [getTranslations('auth.label.password'), 'password'],
        [getTranslations('auth.label.phone'), 'phone'],
      ],
      form: {
        email: '',
        password: '',
        phone: '',
      },
    };
    const resetPwd = {
      label: [
        [getTranslations('auth.label.email'), 'email'],
        [
          getTranslations('auth.label.newpassword'),
          'password',
          getTranslations('auth.label.confirmpassword'),
          'confirmpassword',
        ],
      ],
      form: {
        email: '',
        confirmpassword: '',
        password: '',
        token: '',
      },
    };

    RegisterExtanded = FormValidationHOC(
      Register,
      this.animatedToPreviousStep,
      this.changeAuthType,
      register,
    );
    LoginExtanded = FormValidationHOC(
      Login,
      this.animatedToPreviousStep,
      this.changeAuthType,
      login,
    );
    ResetPwdExtanded = FormValidationHOC(
      ResetPassword,
      this.animatedToPreviousStep,
      this.changeAuthType,
      resetPwd,
    );
  }

  componentWillReceiveProps(nextProps) {
    const { authProvider, authType } = this.state;

    if (
      (this.props.error !== nextProps.error && nextProps.error !== '') ||
      (this.props.fbError !== nextProps.fbError && nextProps.fbError !== '') ||
      (this.props.googleError !== nextProps.googleError &&
        nextProps.googleError !== '')
    ) {
      this.setState(
        state => {
          return {
            ...state,
            isButtonDisabled: false,
          };
        },
        () => {
          Animated.delay(4000).start(() => {
            this.props.resetAllErrors();
          });
        },
      );
    }

    if (nextProps.error || nextProps.fbError || nextProps.googleError) {
      return this.setState(state => ({
        ...state,
        authProvider: '',
      }));
    }
    if (this.props.token !== nextProps.token && nextProps.token !== '') {
      if (authType === 'register') {
        // Check if register with email or social to ask missing infos
        if (authProvider === 'google' || authProvider === 'facebook') {
          // Ask for phone
          this.animatedToNextStep(2);
        } else if (!nextProps.parentProfile.homeAddress) {
          // Go to privacies
          this.navigateToTerms();
        } else {
          this.resetNav('CodeSelector');
          // this.props.navigation.navigate('CodeSelector');
        }
      } else if (authProvider === 'google' || authProvider === 'facebook') {
        // Check if login with social to ask missing infos
        if (
          !nextProps.parentProfile.phone ||
          nextProps.parentProfile.phone === '0'
        ) {
          // Ask for phone
          this.animatedToNextStep(2);
        } else if (!nextProps.parentProfile.homeAddress) {
          // Go to privacies
          this.navigateToTerms();
        } else if (
          !nextProps.parentProfile.securityCode ||
          nextProps.parentProfile.securityCode.length < 4
        ) {
          this.resetNav('CodeSelector');
        } else if (
          Object.keys(nextProps.childProfile).length === 0 ||
          nextProps.childProfile.id === undefined
        ) {
          this.resetNav('Pairing');
        } else {
          this.resetNav('Swiper');
        }
      } else if (
        !nextProps.parentProfile.securityCode ||
        nextProps.parentProfile.securityCode.length < 4
      ) {
        this.resetNav('CodeSelector');
      } else if (
        Object.keys(nextProps.childProfile).length === 0 ||
        nextProps.childProfile.id === undefined
      ) {
        this.resetNav('Pairing');
      } else {
        this.resetNav('Swiper');
      }
    } else if (
      nextProps.token !== '' &&
      authProvider !== '' &&
      !shallowequal(nextProps.parentProfile, this.props.parentProfile) &&
      nextProps.parentProfile &&
      nextProps.parentProfile.phone === '0'
    ) {
      this.animatedToNextStep(2);
    } else if (
      nextProps.token !== '' &&
      authProvider !== '' &&
      !shallowequal(nextProps.parentProfile, this.props.parentProfile) &&
      nextProps.parentProfile &&
      !nextProps.parentProfile.homeAddress
    ) {
      // Go to privacies
      this.navigateToTerms();
    } else if (
      nextProps.token !== '' &&
      authProvider !== '' &&
      !shallowequal(nextProps.parentProfile, this.props.parentProfile) &&
      nextProps.parentProfile &&
      (!nextProps.parentProfile.securityCode ||
        nextProps.parentProfile.securityCode.length < 4)
    ) {
      this.resetNav('CodeSelector');
    }
    return null;
  }

  componentWillUnmount() {
    RegisterExtanded = null;
    LoginExtanded = null;
    ResetPwdExtanded = null;
  }

  resetNav(routeName) {
    const resetAction = NavigationActions.reset({
      index: 0,
      actions: [NavigationActions.navigate({ routeName })],
    });

    this.props.navigation.dispatch(resetAction);
  }

  @bind
  animatedToNextStep(step) {
    const {
      controls,
      visual,
      imageContainer,
      image,
      modalStyle,
      form,
    } = this.state;

    Animated.sequence([
      Animated.parallel([
        Animated.timing(controls.opacity, {
          toValue: 0,
        }),
        Animated.timing(visual.opacity, {
          toValue: 0,
        }),
      ]),
      Animated.parallel([
        Animated.timing(imageContainer.translateY, {
          toValue: step === 1 ? -225 : -Dimensions.get('window').height,
          easing: Easing.elastic(0.8),
        }),
        Animated.timing(image.transform[0].translateY, {
          toValue: 175,
          easing: Easing.elastic(0.8),
        }),
        Animated.timing(controls.height, {
          toValue: step === 1 ? 225 : Dimensions.get('window').height,
          easing: Easing.elastic(0.8),
        }),
        Animated.timing(modalStyle.opacity, {
          toValue: 0.5,
          easing: Easing.elastic(0.8),
        }),
      ]),
    ]).start(() => {
      this.setState(
        state => {
          return {
            ...state,
            step: state.step + 1,
          };
        },
        () => {
          Animated.parallel([
            Animated.timing(controls.opacity, {
              toValue: 1,
            }),
            Animated.timing(form.opacity, {
              toValue: 1,
            }),
          ]).start();
        },
      );
    });
  }

  @bind
  animatedToPreviousStep() {
    const {
      controls,
      visual,
      imageContainer,
      image,
      modalStyle,
      form,
    } = this.state;

    this.setState(
      state => {
        return {
          ...state,
          animRunning: true,
        };
      },
      () => {
        Animated.parallel([
          Animated.timing(form.opacity, {
            toValue: 0,
          }),
          Animated.timing(controls.opacity, {
            toValue: 0,
          }),
        ]).start(() => {
          this.setState(
            state => {
              return {
                ...state,
                step: state.step - 1,
              };
            },
            () => {
              const { step } = this.state;

              Animated.sequence([
                Animated.parallel([
                  Animated.timing(modalStyle.opacity, {
                    toValue: step === 1 ? 0.5 : 0,
                    easing: Easing.elastic(0.6),
                  }),
                  Animated.timing(imageContainer.translateY, {
                    toValue: step === 0 ? 0 : -225,
                    easing: Easing.elastic(0.6),
                  }),
                  Animated.timing(image.transform[0].translateY, {
                    toValue: step === 0 ? 0 : 175,
                    easing: Easing.elastic(0.6),
                  }),
                  Animated.timing(visual.opacity, {
                    toValue: 1,
                    easing: Easing.elastic(0.6),
                  }),
                  Animated.timing(controls.height, {
                    toValue: step === 0 ? 0 : 225,
                    easing: Easing.elastic(0.6),
                  }),
                ]),
                Animated.timing(controls.opacity, {
                  toValue: 1,
                }),
              ]).start(() => {
                this.setState(state => {
                  return {
                    ...state,
                    animRunning: false,
                  };
                });
              });
            },
          );
        });
      },
    );
  }

  @bind
  selectAuthType(authType) {
    if (
      this.props.error !== '' ||
      this.props.fbError !== '' ||
      this.props.googleError !== ''
    ) {
      this.props.resetAllErrors();
    }

    this.setState(
      state => {
        return {
          ...state,
          authType,
        };
      },
      () => {
        this.animatedToNextStep(1);
      },
    );
  }

  @bind
  changeAuthType(authType = 'register') {
    this.props.resetAllErrors();

    Animated.timing(this.state.form.opacity, {
      toValue: 0,
      duration: 300,
    }).start(() => {
      this.setState(
        state => {
          return {
            ...state,
            authType,
          };
        },
        () => {
          Animated.timing(this.state.form.opacity, {
            toValue: 1,
            duration: 300,
          }).start();
        },
      );
    });
  }

  @bind
  selectAuthProvider(authProvider) {
    this.setState(state => {
      return {
        ...state,
        authProvider,
        isButtonDisabled: true,
      };
    });
  }

  @bind
  navigateToTerms(direction = 'forward') {
    const { form: { opacity }, authProvider } = this.state;
    let nextStep = 0;

    if (direction === 'forward') {
      nextStep = 4;
    } else if (authProvider === 'facebook' || authProvider === 'google') {
      nextStep = 1;
    } else {
      nextStep = 2;
    }

    Animated.timing(opacity, {
      toValue: 0,
    }).start(() => {
      this.setState(
        state => {
          return {
            ...state,
            step: nextStep,
            authProvider: state.authProvider !== '' ? '' : state.authProvider,
            isButtonDisabled: false,
          };
        },
        () => {
          Animated.timing(opacity, {
            toValue: 1,
          }).start();
        },
      );
    });
  }

  renderForms() {
    const { authType, form, authProvider } = this.state;
    const { error, logginIn } = this.props;

    if (authType === 'login') {
      return (
        <LoginExtanded
          errorApi={error}
          styling={form}
          changeAuthType={this.changeAuthType}
          authProvider={authProvider}
          logginIn={logginIn}
        />
      );
    } else if (authType === 'register') {
      return (
        <RegisterExtanded
          errorApi={error}
          styling={form}
          authProvider={authProvider}
          logginIn={logginIn}
          navigateToTerms={this.navigateToTerms}
        />
      );
    }

    return (
      <ResetPwdExtanded
        errorApi={error}
        styling={form}
        changeAuthType={this.changeAuthType}
      />
    );
  }

  render() {
    const {
      fbError,
      googleError,
      navigation,
      partialRegister,
      parentProfile,
      profileError,
    } = this.props;
    const {
      authType,
      isButtonDisabled,
      authProvider,
      step,
      imageContainer,
      controls,
      visual,
      controlsShown,
      modalStyle,
      image,
      animRunning,
      form: { opacity },
    } = this.state;

    return (
      <StyledContainerBlack>
        <Styles.StyledWrapperAuth style={appStyles.cardStyle}>
          {step === 0 || step === 1 ? ( // eslint-disable-line
            <StyledAndroidWrapper>
              <Styles.AnimatedOverflow
                style={{
                  transform: [{ translateY: imageContainer.translateY }],
                }}
              >
                <Visual
                  step={step}
                  animRunning={animRunning}
                  controls={controlsShown}
                  modalStyle={modalStyle}
                  imageStyle={image}
                  styling={visual}
                  selectAuthType={this.selectAuthType}
                  animatedToPreviousStep={this.animatedToPreviousStep}
                />
              </Styles.AnimatedOverflow>
              {step === 1 && (
                <SelectAuthProvider
                  authType={authType}
                  isButtonDisabled={isButtonDisabled}
                  authProvider={authProvider}
                  styling={controls}
                  fbError={fbError}
                  googleError={googleError}
                  goToNextStep={this.animatedToNextStep}
                  registerWithGoogle={this.registerWithGoogle}
                  loginWithGoogle={this.loginWithGoogle}
                  registerWithFb={this.registerWithFb}
                  loginWithFb={this.loginWithFb}
                  selectAuthProvider={this.selectAuthProvider}
                />
              )}
            </StyledAndroidWrapper>
          ) : step !== 4 ? (
            this.renderForms()
          ) : (
            <TermsOfServices
              navigation={navigation}
              opacity={opacity}
              partialRegister={partialRegister}
              startRegister={this.startRegister}
              parentProfile={parentProfile}
              editUserProfile={this.editUserProfile}
              profileError={profileError}
              navigateToTerms={this.navigateToTerms}
            />
          )}
        </Styles.StyledWrapperAuth>
      </StyledContainerBlack>
    );
  }
}

Auth.propTypes = {
  logginIn: bool.isRequired,
  navigation: shape({
    dispatch: func.isRequired,
    goBack: func.isRequired,
    navigate: func.isRequired,
    setParams: func.isRequired,
    state: shape({
      key: string.isRequired,
      routeName: string.isRequired,
    }).isRequired,
  }).isRequired,
  fbError: string.isRequired,
  googleError: string.isRequired,
  error: string.isRequired,
  token: string.isRequired,
  resetAllErrors: func.isRequired,
  registerWithGoogle: func.isRequired,
  loginWithGoogle: func.isRequired,
  registerWithFb: func.isRequired,
  loginWithFb: func.isRequired,
  parentProfile: shape({
    phone: string,
    securityCode: string,
  }).isRequired,
  childProfile: shape({
    nickname: string,
  }).isRequired,
  partialRegister: shape({
    email: string,
    firstname: string,
    lastname: string,
    phone: string,
    password: string,
    language: string,
    birthday: string,
    homeAddress: shape({
      country: string,
      address: string,
      postalCode: string,
      city: string,
      region: string,
    }),
  }).isRequired,
  startRegister: func.isRequired,
  editUserProfile: func.isRequired,
  profileError: string.isRequired,
};

export default Auth;
