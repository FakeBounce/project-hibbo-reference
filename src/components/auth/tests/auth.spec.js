import React from 'react';
import renderer from 'react-test-renderer';
import EStyleSheet from 'react-native-extended-stylesheet';

import appStyles from 'styles/appStyles';

import Auth from '../components/Auth';

EStyleSheet.build(appStyles);

describe('Auth', () => {
  const fn = jest.fn();
  const navigation = {
    dispatch: () => {},
    goBack: () => {},
    navigate: () => {},
    setParams: () => {},
    state: {
      key: '',
      routeName: '',
    },
  };

  const component = renderer.create(
    <Auth
      startLogin={fn}
      startRegister={fn}
      resetPwdStart={fn}
      registerWithGoogle={fn}
      loginWithGoogle={fn}
      registerWithFb={fn}
      loginWithFb={fn}
      navigation={navigation}
      token=""
      fbError=""
      googleError=""
      error=""
      resetAllErrors={() => {}}
    />,
  ).toJSON();

  it('should render the component', () => {
    expect(component).toMatchSnapshot();
  });
});
