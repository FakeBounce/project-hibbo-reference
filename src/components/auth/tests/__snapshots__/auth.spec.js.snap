// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth should render the component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 1,
    }
  }
>
  <View>
    <View
      collapsable={undefined}
      style={
        Object {
          "overflow": "hidden",
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Image
          collapsable={undefined}
          source={1}
          style={
            Object {
              "height": 1334,
              "opacity": 1,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
              "width": 750,
            }
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "flex-end",
                "marginBottom": 50,
                "opacity": 1,
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#FFF",
                      "fontFamily": "Circular-Std-Medium",
                    },
                    Object {
                      "fontSize": 35,
                      "marginBottom": 15,
                    },
                  ]
                }
              >
                Register
              </Text>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#FFF",
                      "fontFamily": "Circular-Std-Medium",
                    },
                    Object {
                      "fontSize": 17,
                    },
                  ]
                }
              >
                Login
              </Text>
            </View>
          </View>
        </Image>
      </View>
    </View>
  </View>
</View>
`;
